### ISSUE TITLE
Nteract Mathjax package has trouble with rendering in Atom

### ISSUE THREAD

**Issue by wadethestealth** [nteract/archived-desktop-app#4461]
### Background Info
We have recently just upgraded all of hydrogen's old components to the new outputs package, which is a great improvement. However, it seems like the @nteract/mathjax does not like this new style of rendering components. When trying to use LaTeX through either python or our new markdown result bubbles, we are constantly getting a Content Security Policy Error.

I have done some debugging, and have tried to figure out the issue, but nothing seems to work.

### Abstract Model of our Result View Component
```html
<MathJax.Provider src={LocalMathjaxFile} > <!-- This is technically in a separate file wrapping our component -->
    <some random and nested divs>
        <Outputs>
            <DisplayData>
                <Media.Markdown />
            </DisplayData>
        </Outputs>
    </some random and nested divs>
</MathJax.Provider >
```

### Abstract Model of a Result View Component that gets rendered
```html
<MathJax.Provider src={LocalMathjaxFile} >
    <some random and nested divs>
        <Outputs>
            <DisplayData>
                <Media.Markdown >
                    <MathJax.Provider src={cdn-that-can't-be-loaded-in-atom} >
                        <MathJax.Node >
                        <MathJax.Node />
                    <MathJax.Provider />
                </Media.Markdown>
            </DisplayData>
        </Outputs>
    </some random and nested divs>
</MathJax.Provider >
```

### The Issues
It seems that the Media.Markdown component is not recognizing or doesn't care about the wrapping provider, and it simply creates a new Provider every time but tries to load from a cdn, which is why I get the CSP error. However, we need it to use the local Provider. Interestingly though, It will still render but only the first markdown cell you run. The second run cell will just be empty, like its waiting on the script to render it. If you use a python code cell, and use the ipython latex, it still gives you the CSP error, but it renders everytime, which I don't understand. On top of this, we are creating providers for every cell result, which results in a bunch of the same MathJax script for every result, and I was wondering if there is a way to fix these issues and only how one script being added to the head of the atom editor. Thanks for any insight you can give @rgbkrk

**Comment by rgbkrk** [comment]
Oh no, that is troubling. It seems the provider logic is off at the moment. Maybe the right approach is to override the Markdown package for now within Hydrogen (set it up to handle the same media type). cc @BenRussert

**Comment by wadethestealth** [comment]
@rgbkrk I have absolutely zero idea what has changed in 2 weeks, however the above result is no longer what we get. Now the problem lies in queueing MathJax to be rerendered. I have no idea when or where to call the [mathjax queue](https://docs.mathjax.org/en/latest/advanced/typeset.html). What I know is that we are left with MathJax Script tags that need to be called, which the docs say you can do, but again I don't know where I should do this in atom/react.

```html
<div class="hydrogen_cell_display" style="max-height: 600px; overflow-y: auto;">
   <div class="markdown">
      <span>
         <span class="mjx-chtml MJXc-display MJXc-processing">
            <span id="MathJax-Element-3-Frame" class="mjx-chtml MathJax_CHTML" tabindex="0">
            </span>
         </span>
         <script type="math/tex; mode=display" id="MathJax-Element-3">    f(x) = f\xi</script>
      </span>
   </div>
</div>
```
Seems like the script tag has an id that just increments everytime we run. So Mathjax should have no problem finding the tag if given it via the queue.

**Comment by wadethestealth** [comment]
It seems like on first run the script seems to think it is processed and the element says processing
I can easily rerender everything, and it looks normal. No just need to figure out where to patch this script in
```javascript
let elementJaxs = MathJax.Hub.getAllJax();
for (let script of elementJaxs) {
   MathJax.Hub.Queue(["Reprocess", script]);
}
```

**Comment by rgbkrk** [comment]
Thank you for continuing to come back to this. I've been a bit overstretched and haven't been able to catch up on this 😭

**Comment by wadethestealth** [comment]
@rgbkrk it seems like the only rendering error is after the very first process, all other elements will be queued as processing and never finish. A reprocess request fixing any unloaded latex, but that's not ideal. I still need to debug why the first request is always left processing. Also it seemed when we created our own markdown component for a different feature, we got the mathjax to use the correct file, which I mentioned as the change that I had no idea the cause of.

**Comment by wadethestealth** [comment]
@rgbkrk Adding a 10 ms wait to the typeset fixes our issue on hydrogen. I am just not sure what we need to assert to remove the wait. 
```
componentDidMount() {
   setTimeout(() => this.typeset(), 10);
}
```

**Comment by rgbkrk** [comment]
Gross yet workable. Well done.

**Comment by wadethestealth** [comment]
@rgbkrk However this isn't implemeted because the code belongs inside our mathjax package

**Comment by wadethestealth** [comment]
@rgbkrk Would the above code be allowed to be merged into the nteract mathjax package?

**Comment by rgbkrk** [comment]
I think so. Please feel free to submit it. I'm finally setting aside some time on the weekend I can catch up with nteract contributions. We'll get this in shape together.
